src/enzo/ActiveParticle_RadiationParticle.C

```diff
@@ -0,0 +1,524 @@
+/***********************************************************************
+
+  
+
+  written by: John Regan
+  date:       March, 2014
+  modified1: John Regan - ported to Enzo-3.0
+
+  PURPOSE: Somewhat idealised treatment of radiation sources. 
+           This routine creates massless radiation sources 
+           at predefined points in the simulation. The sources
+           emit radiation for a user defined time. 
+          
+ 
+  RETURNS:
+     SUCCESS if the particle is successfully created.
+     FAIL otherwise
+***********************************************************************/
+
+#include "ActiveParticle_RadiationParticle.h"
+#include "phys_constants.h"
+#include "CosmologyParameters.h"
+
+
+#define APDEBUG 0
+#define NUMPARAMS 4
+#define RadiationSourceLifeTime 1e10   //i.e. infinite
+
+/* We need to make sure that we can operate on the grid, so this dance is
+ * necessary to make sure that grid is 'friend' to this particle type. */
+
+class RadiationParticleGrid : private grid {
+  friend class ActiveParticleType_RadiationParticle;
+};
+
+//Initialize static variables
+int ActiveParticleType_RadiationParticle::RadiationNumSources          = INT_UNDEFINED;
+InitData *ActiveParticleType_RadiationParticle::Root                   = NULL;
+int ActiveParticleType_RadiationParticle::RadiationSEDNumberOfBins     = INT_UNDEFINED;
+float* ActiveParticleType_RadiationParticle::RadiationEnergyBins       = NULL;
+float* ActiveParticleType_RadiationParticle::RadiationSED              = NULL;
+float ActiveParticleType_RadiationParticle::RadiationLifetime          = FLOAT_UNDEFINED;
+float ActiveParticleType_RadiationParticle::RadiationPhotonsPerSecond  = FLOAT_UNDEFINED;
+bool ActiveParticleType_RadiationParticle::FixedInSpace                = TRUE;
+//Redshift
+float CR = 0.0;
+
+int ActiveParticleType_RadiationParticle::InitializeParticleType() {
+
+  RadiationNumSources = NumberOfRadiationParticles;
+  RadiationSEDNumberOfBins = NumberOfEnergyBins;
+  RadiationEnergyBins = new float[RadiationSEDNumberOfBins];
+  RadiationSED = new float[RadiationSEDNumberOfBins];
+  for(int i = 0; i < RadiationSEDNumberOfBins; i++) {
+    RadiationEnergyBins[i] = RadiationEnergyInBin[i];
+    RadiationSED[i] = RadiationBinSED[i];
+  }
+  // Example of what values may look like
+  // RadiationEnergyBins[1] = 25.0;
+  // RadiationSED[1] = 100.0/RadiationSEDNumberOfBins;
+  // RadiationEnergyBins[2] = 55.0;
+  // RadiationSED[2] = 100.0/RadiationSEDNumberOfBins;
+  // RadiationEnergyBins[3] = 12.8;
+  // RadiationSED[3] = 100.0/RadiationSEDNumberOfBins;
+  // RadiationEnergyBins[4] = 1;
+  // RadiationSED[4] = 100.0/RadiationSEDNumberOfBins;
+  // RadiationEnergyBins[5] = 200.0;
+  // RadiationSED[5] = 100.0/RadiationSEDNumberOfBins;
+  RadiationLifetime = RadiationSourceLifeTime;
+  RadiationPhotonsPerSecond = PhotonsPerSecond;
+  FixedInSpace = TRUE; //Hardcoded for the moment but could potentially be a parameter
```

Comment:
It should be made into a parameter before merging.

Reply:
I'm not so sure. The RadiationParticle is fixed in space in this implementation. Adding another parameter isn't necessarily a good idea. My view is that we leave this as is and if someone wants to use the RadiationParticle and allow it to advect in the future then they can modify this.